### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by rjw57**
This PR provides an initial implementation of the staff on-costs calculator.

The module is verified by checking that it can reproduce exactly the tables listed on the HR website. Since those tables are private, we cannot simply include the tables in the test suite. Instead we include their SHA256 hash and verify that the tables produced from our code have the correct hash.

If you have the tables to hand, you can add them back into the .../test_data/private directory as outlined in the README there. If the CSV files are present, the test suite will give more useful information in the case of a discrepancy.

This PR includes a basic CircleCI configuration which will run the tests, submit code coverage, build the documentation and, when a commit lands on master, update the documentation hosted via GitHub pages.

An example of the documentation is viewable at: https://rjw57.github.io/pidash-ucamstaffoncosts/.

Closes uisautomation/pidash-webapp#52.

### CODE REVIEW

File: ucamstaffoncosts/tax.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import fractions
```
Comment (msb) [inline ucamstaffoncosts/tax.py:1]: consider splitting this into nic and apprenticeship modules

File: ucamstaffoncosts/tax.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return contribution
+
+    return calculate
+
+
+#: Table A employer NICs keyed by tax year.
+TABLE_A_EMPLOYER_NIC = {
+    2018: _make_nic_calculator((
+        (6032, 0),
+
```
Comment (msb) [inline ucamstaffoncosts/tax.py]: remove this space

**Review by msb — COMMENTED** [review]
all looks good - as far as I can see

File: ucamstaffoncosts/tax.py (hunk 2 above)
Comment (rjw57) [reply ucamstaffoncosts/tax.py]: I believe the convention for ``dict`` literals is to have a space between the key and the value. I just checked with a trivial file:
```
# test.py
{
    'foo':'bar'
}
```
and flake8 complains:
```
test.py:3:10: E231 missing whitespace after ':'
```

File: ucamstaffoncosts/tax.py (hunk 1 above)
Comment (rjw57) [reply ucamstaffoncosts/tax.py:1]: I'm somewhat neutral on this. To me it make sense that all tax calculations happen in the same module but I'm not going to scream about having them split. If it's truly just "consider", then I've considered and declined. If it's "do it or I'll not approve", I'll do it.

File: ucamstaffoncosts/tax.py (hunk 2 above)
Comment (rjw57) [reply ucamstaffoncosts/tax.py]: wilco

File: ucamstaffoncosts/tax.py (hunk 1 above)
Comment (msb) [reply ucamstaffoncosts/tax.py:1]: just consider :)

File: ucamstaffoncosts/tax.py (hunk 2 above)
Comment (rjw57) [reply ucamstaffoncosts/tax.py]: Addressed in 03bf9aa.

File: ucamstaffoncosts/tax.py (hunk 1 above)
Comment (msb) [reply ucamstaffoncosts/tax.py]: Tested by manually plugging in different values for different schemes and comparing them to the HR tables.

**Comment by msb** [comment]
Tested by manually plugging in different values for different schemes and comparing them to the HR tables.

**Pull request merged**